tests/pkg_c25519/app.config.test

```diff
@@ -3,5 +3,5 @@ CONFIG_MODULE_RANDOM=y
 # Should be autoselecting the MODULE_PRNG_HWRNG if possible
 # Since the makefile cannot we have to override until end of migration
 # Remove when TEST_KCONFIG is complete
-CONFIG_MODULE_PRNG_TINYMT32=y
+CONFIG_MODULE_PRNG_MUSL_LCG=y
```

Comment:
Neither Tinymt32, nor LCG are crypto RNGs. One step further, HWRNGs are not necessarily secure for direct use. Hence, we should default to a CSPRNG in crypto contexts. `SHA256PRNG` would be the leanest choice.

This shows the problem with our current `random` module: There is only a single (non-secure) default RNG. As said, I am planning to continue my work on that starting from #15671.

Reply:
What I meant: crypto packages (cifra, hacl, dsa, ...) that require access to a random number generator should configure a crypto RNG (SHA256PRNG)

Reply:
I agree, but let's do this as a follow-up. This PR only changes the default RNG, those apps are still using the default RNG  - if they should not, that would be a different PR.

Reply:
What if we say "set correct default choice"? Anyway, we can also go with two PRs.